### PR TITLE
docs(contributing): Adding docs/developer.md page

### PR DESCRIPTION
**Pull request by niladrih**
**Why is this PR required? What issue does it fix?**:
This PR adds the page docs/developer.md. This also fixes the dead link in the CONTRIBUTING.md page which links docs/developer.md

**What this PR does?**:
The developer.md page includes instructions for setting up the development environment.

~~**Does this PR require any upgrade changes?**:~~

~~**If the changes in this PR are manually verified, list down the scenarios covered:**:~~

Signed-off-by: Niladri Halder <niladri.halder@mayadata.io>

### CODE REVIEW

File: docs/developer.md [hunk 1]
```diff
@@ -0,0 +1,205 @@
...
+
+- [Prerequisites](#prerequisites)
+- [Initial Setup](#initial-setup)
+- [Development](#development)
+- [Dependencies](#dependencies)
+- [Submitting Your Changes](#submitting-your-changes)
+
+## Prerequisites
+
+* You have Go 1.10+ installed on your local host/development machine.
```
Comment (akhilerm) [inline docs/developer.md]: Go version that is used in the [build file](https://github.com/openebs/dynamic-localpv-provisioner/blob/develop/buildscripts/provisioner-localpv/provisioner-localpv.Dockerfile#L15) should be used.

File: docs/developer.md [hunk 2]
```diff
@@ -0,0 +1,205 @@
...
+* You have Docker installed on your local host/development machine. Docker is required for building maya component container images and to push them into a Kubernetes cluster for testing.
```
Comment (akhilerm) [inline docs/developer.md]: replace maya  components with localpv

File: docs/developer.md [hunk 3]
```diff
@@ -0,0 +1,205 @@
...
+### Always sync your local repository
+
+Open a terminal on your local host. Change directory to the maya-fork root.
+
+```sh
+
+$ cd path/to/directory/openebs/dynamic-localpv-provisioner.git
+```
+
+ Checkout the master branch.
```
Comment (akhilerm) [inline docs/developer.md]:
```suggestion
 Checkout the develop branch.
```

File: docs/developer.md [hunk 4]
```diff
@@ -0,0 +1,205 @@
...
+
+ ```sh
+ $ git checkout develop
+ Switched to branch 'develop'
+ Your branch is up-to-date with 'origin/develop'.
+ ```
+
+ Recall that origin/develop is a branch on your remote GitHub repository.
+ Make sure you have the upstream remote openebs/maya by listing them.
```
Comment (akhilerm) [inline docs/developer.md]: replace maya with localpv repo name.

File: docs/developer.md [hunk 5]
```diff
@@ -0,0 +1,205 @@
...
+ upstream no_push (push)
+ ```
+
+ If the upstream is missing, add it by using below command.
+
+ ```sh
+ $ git remote add upstream https://github.com/openebs/dynamic-localpv-provisioner.git
+ ```
+
+ Fetch all the changes from the upstream master branch.
```
Comment (akhilerm) [inline docs/developer.md]:
```suggestion
 Fetch all the changes from the upstream develop branch.
```

File: docs/developer.md [hunk 6]
```diff
@@ -0,0 +1,205 @@
...
+
+ Rebase your local master with the upstream/master.
+
+ ```sh
+ $ git rebase upstream/develop
+ First, rewinding head to replay your work on top of it...
+ Fast-forwarded master to upstream/develop.
+ ```
+
+ This command applies all the commits from the upstream master to your local master.
```
Comment (akhilerm) [inline docs/developer.md]:
```suggestion
 This command applies all the commits from the upstream develop to your local develop.
```

File: docs/developer.md [hunk 7]
```diff
@@ -0,0 +1,205 @@
...
+ $ git status
+ On branch develop
+ Your branch is ahead of 'origin/develop' by 38 commits.
+ (use "git push" to publish your local commits)
+ nothing to commit, working directory clean
+ ```
+
+ Your local repository now has all the changes from the upstream remote. You need to push the changes to your own remote fork which is origin develop.
+
+ Push the rebased master to origin develop.
```
Comment (akhilerm) [inline docs/developer.md]:
```suggestion
 Push the rebased develop to origin develop.
```

File: docs/developer.md [hunk 8]
```diff
@@ -0,0 +1,205 @@
...
+git rebase upstream/develop
+```
+
+While you rebase your changes, you must resolve any conflicts that might arise and build and test your changes using the above steps.
+
+### Dependencies
+
+#### Go modules
+
+This repo supports [Go Modules](https://github.com/golang/go/wiki/Modules), and
```
Comment (akhilerm) [inline docs/developer.md]: Instead of support,  we now use go modules only.

**Review by akhilerm — CHANGES_REQUESTED** [review]
Some changes are required regarding:
1. changing branch to develop from master
2. changing repo name in the docs.

**Comment by niladrih** [comment]
Hi @akhilerm . I amended the previous commit and did a 'Resolve conversation' on the suggested changes. Also I re-requested a review from you. 

Let me know if any of this violates best practices.

**Comment by akhilerm** [comment]
@niladrih instead of amending the commits and force pushing., you can just push as new commits. This will help to review only the changes done.  The rest is good.

**Review by akhilerm — APPROVED** [review]
LGTM

**Review by kmova — APPROVED** [review]

**Pull request merged**